hardhat.config.ts

```diff
@@ -1,14 +1,50 @@
-/**
- * @type import('hardhat/config').HardhatUserConfig
- */
-module.exports = {
+import '@nomiclabs/hardhat-waffle'
+import {existsSync, readFileSync} from 'fs'
+import {HardhatUserConfig} from 'hardhat/config'
+import {NetworkUserConfig} from 'hardhat/types'
+
+let infuraApiKey: string, mnemonic: string
+
+if (!existsSync('.infuraKey')) {
+  throw new Error("Put infura api key inside a .infuraKey file, don't paste it in the config.") // infura api key
+}
```

Comment:
I think it's better to read infura key as env variable (from `.env`), cause we will have etherscan api key, coinmarket cap api key..etc, using a separate file for each one is kind of dumb, we will also need to update gitignore everything we add a new api key.